src/flowcean/transforms/one_cold.py

```diff
@@ -0,0 +1,153 @@
+
+import polars as pl
+
+from flowcean.core.transform import Transform
+from flowcean.transforms.one_hot import (
+    NoCategoriesError,
+    NoMatchingCategoryError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class OneCold(Transform):
+    """Transforms integer features into a set of binary one-cold features.
+
+    Transforms integer features into a set of binary one-cold features. The
+    original integer features are dropped and are not part of the resulting
+    data frame.
+
+    As an example consider the following data
+
+    feature |
+    :------:|
+     0      |
+     1      |
+     2      |
+     1      |
+     5      |
+
+    When the one-cold transformation is applied, the result is as follows
+
+    feature_0 | feature_1 | feature_2 | feature_5
+    ----------|-----------|-----------|----------
+     0        | 1         | 1         | 1
+     1        | 0         | 1         | 1
+     1        | 1         | 0         | 1
+     1        | 0         | 1         | 1
+     1        | 1         | 1         | 0
+    """
+
+    feature_category_mapping: dict[str, dict[str, Any]]
+    check_for_missing_categories: bool
+
+    def __init__(
+        self,
+        feature_categories: dict[str, list[Any]],
+        *,
+        check_for_missing_categories: bool = False,
+    ) -> None:
+        """Initializes the One-Hot transform.
+
+        Args:
+            feature_categories: Dictionary of features and a list of
+                categorical values to encode for each.
+            check_for_missing_categories: If set to true, a check is performed
+                to see if all values belong to a category. If an unknown value
+                is found which does not belong to any category, a
+                NoMatchingCategoryError is thrown. To perform this check, the
+                dataframe must be materialised, resulting in a potential
+                performance decrease. Therefore it defaults to false.
+        """
+        self.feature_category_mapping = {
+            feature: {f"{feature}_{value}": value for value in values}
+            for feature, values in feature_categories.items()
+        }
+        self.check_for_missing_categories = check_for_missing_categories
+
+    @override
+    def apply(
+        self,
+        data: pl.DataFrame,
+    ) -> pl.DataFrame:
+        """Transform data with this one hot transformation.
+
+        Transform data with this one hot transformation and return the
+        resulting dataframe.
+
+        Args:
+            data: The data to transform.
+
+        Returns:
+            The transformed data.
+        """
+        if len(self.feature_category_mapping) == 0:
+            raise NoCategoriesError
+        for (
+            feature,
+            category_mappings,
+        ) in self.feature_category_mapping.items():
+            data = data.with_columns(
+                [
+                    pl.col(feature).ne(value).cast(pl.Int64).alias(name)
```

Comment:
As the polars expressions here and in lines 100-107 are quite hard to grasp, I am wondering whether it's worth to add some inline comments or extend the doc-string?